src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8493,175 +8502,83 @@ protected MethodGroupResolution ResolveExtension(
-                    if (extensionMethodResult.HasAnyApplicableMethod)
-                    {
-                        firstResult.Free();
-                        actualArguments?.Free();
-                        return extensionMethodResult;
-                    }
-                    else if (firstResult.IsEmpty)
-                    {
-                        firstResult = extensionMethodResult;
-                    }
-                    else
-                    {
-                        // Neither the first result, nor applicable. No need to save result.
-                        extensionMethodResult.Free();
-                    }
-                }
+                lookupOptions |= LookupOptions.MustBeInvocableIfMember;
             }
 
-            Debug.Assert((actualArguments == null) || !firstResult.IsEmpty);
-            actualArguments?.Free();
-
-            return firstResult;
-
-            static bool tryResolveExtensionMember(Binder binder, SyntaxNode expression, string memberName, AnalyzedArguments? analyzedArguments, BoundExpression left,
-                ImmutableArray<TypeWithAnnotations> typeArgumentsWithAnnotations, OverloadResolution.Options options, RefKind returnRefKind, in CallingConventionInfo callingConvention,
-                TypeSymbol? returnType, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo, ExtensionScope scope, out MethodGroupResolution extensionResult)
+            foreach (var scope in new ExtensionScopes(this))
             {
-                var lookupResult = LookupResult.GetInstance();
-
-                int arity = typeArgumentsWithAnnotations.IsDefault ? 0 : typeArgumentsWithAnnotations.Length;
-                var lookupOptions = (arity == 0) ? LookupOptions.AllMethodsOnArityZero : LookupOptions.Default;
-                if (analyzedArguments is not null)
-                {
-                    lookupOptions |= LookupOptions.MustBeInvocableIfMember;
-                }
+                lookupResult.Clear();
+                classicExtensionLookupResult.Clear();
 
                 scope.Binder.LookupExtensionMembersInSingleBinder(
                     lookupResult, left.Type!, memberName, arity,
-                    basesBeingResolved: null, lookupOptions, originalBinder: binder, ref useSiteInfo);
+                    basesBeingResolved: null, lookupOptions, originalBinder: this, ref useSiteInfo);
 
-                if (!lookupResult.IsMultiViable)
-                {
-                    lookupResult.Free();
-                    extensionResult = default;
-                    return false;
-                }
+                var diagnostics = BindingDiagnosticBag.GetInstance(withDiagnostics: true, useSiteInfo.AccumulatesDependencies);
 
-                var members = ArrayBuilder<Symbol>.GetInstance();
+                CompoundUseSiteInfo<AssemblySymbol> classicExtensionUseSiteInfo = GetNewCompoundUseSiteInfo(diagnostics);
```

Comment:
:memo: this is irregular (difference between classic and new extension methods). From offline discussion the new behavior (report all use-site diagnostics) is better, but we also don't want to risk impacting the behavior of classic extension methods...